pkg/proc/core/minidump/minidump.go

```diff
@@ -0,0 +1,689 @@
+		module.SizeOfImage = buf.u32()
+		module.Checksum = buf.u32()
+		module.TimeDateStamp = buf.u32()
+		nameOff := int(buf.u32())
+
+		versionInfoVec := make([]uint32, unsafe.Sizeof(VSFixedFileInfo{})/unsafe.Sizeof(uint32(0)))
+		for j := range versionInfoVec {
+			versionInfoVec[j] = buf.u32()
+		}
+
+		module.VersionInfo = *(*VSFixedFileInfo)(unsafe.Pointer(&versionInfoVec[0]))
+
+		_, module.CVRecord = readLocationDescriptor(buf)
+		_, module.MiscRecord = readLocationDescriptor(buf)
+
+		if buf.err != nil {
+			return
+		}
+
+		nameBuf := minidumpBuf{buf: buf.buf, kind: "file", off: nameOff, err: nil, ctx: buf.ctx}
+		module.Name = readString(&nameBuf)
+		if nameBuf.err != nil {
+			buf.err = nameBuf.err
+			return
+		}
+
+	}
+}
+
+// readMemory64List reads a _MINIDUMP_MEMORY64_LIST structure, containing
+// the description of the process memory.
+// See: https://docs.microsoft.com/en-us/windows/desktop/api/minidumpapiset/ns-minidumpapiset-_minidump_memory64_list
+// And: https://docs.microsoft.com/en-us/windows/desktop/api/minidumpapiset/ns-minidumpapiset-_minidump_memory_descriptor
+func readMemory64List(mdmp *Minidump, buf *minidumpBuf, logfn func(fmt string, args ...interface{})) {
+	rangesNum := buf.u64()
+	baseOff := int(buf.u64())
+	if buf.err != nil {
+		return
+	}
+
+	_ = baseOff
+
+	for i := uint64(0); i < rangesNum; i++ {
+		addr := buf.u64()
+		sz := buf.u64()
+
+		end := baseOff + int(sz)
+		if baseOff >= len(buf.buf) || end > len(buf.buf) {
+			buf.err = fmt.Errorf("memory range at %#x of size %#x is past the end of file, while %s", baseOff, sz, buf.ctx)
+			return
+		}
+
+		mdmp.addMemory(addr, buf.buf[baseOff:end])
+
+		if logfn != nil {
+			logfn("\tMemory %d addr:%#x size:%#x FileOffset:%#x\n", i, addr, sz, baseOff)
+		}
+
+		baseOff = end
+	}
+}
+
+func readMemoryInfoList(mdmp *Minidump, buf *minidumpBuf, logfn func(fmt string, args ...interface{})) {
+	startOff := buf.off
+	sizeOfHeader := int(buf.u32())
+	sizeOfEntry := int(buf.u32())
+	numEntries := buf.u64()
+
+	buf.off = startOff + sizeOfHeader
+
+	mdmp.MemoryInfo = make([]MemoryInfo, numEntries)
+
+	for i := range mdmp.MemoryInfo {
+		memInfo := &mdmp.MemoryInfo[i]
+		startOff := buf.off
+
+		memInfo.Addr = buf.u64()
+		buf.u64() // allocation_base
+
+		buf.u32() // allocation_protection
+		buf.u32() // alignment
+
+		memInfo.Size = buf.u64()
+
+		memInfo.State = MemoryState(buf.u32())
+		memInfo.Protection = MemoryProtection(buf.u32())
+		memInfo.Type = MemoryType(buf.u32())
+
+		if logfn != nil {
+			logfn("\tMemoryInfo %d Addr:%#x Size:%#x %s %s %s\n", i, memInfo.Addr, memInfo.Size, memInfo.State, memInfo.Protection, memInfo.Type)
+		}
+
+		buf.off = startOff + sizeOfEntry
+	}
+}
+
+// readMiscInfo reads the process_id from a MiscInfo stream.
+func readMiscInfo(mdmp *Minidump, buf *minidumpBuf) {
+	buf.u32() // size of info
+	buf.u32() // flags1
+
+	mdmp.Pid = buf.u32() // process_id
+	// there are more fields here, but we don't care about them
```

Comment:
Should we move the buffer past this memory, or are we simply done with the stream at this point?

Reply:
We're done. The buffer used here is for the misc info stream only.